src/Agent.Worker/Worker.cs

```diff
@@ -102,6 +103,61 @@ public async Task<int> RunAsync(string pipeIn, string pipeOut)
             }
         }
 
+        private string ScrubPiiData(Pipelines.AgentJobRequestMessage message)
+        {
+            ArgUtil.NotNull(message, nameof(message));
+
+            var scrubbedVariables = new Dictionary<string, VariableValue>();
+
+            // Scrub the known PII variables
+            foreach (var variable in message.Variables)
+            {
+                if (Variables.PiiVariables.Contains(variable.Key))
+                {
+                    scrubbedVariables[variable.Key] = "***";
+                }
+                else
+                {
+                    scrubbedVariables[variable.Key] = variable.Value;
+                }
+            }
+
+            var scrubbedRepositories = new List<Pipelines.RepositoryResource>();
+
+            // Scrub the repository resources
+            foreach (var repository in message.Resources.Repositories)
+            {
+                Pipelines.RepositoryResource scrubbedRepository = repository.Clone();
+
+                scrubbedRepository.Properties.Set("VersionInfo", "***");
```

Comment:
there is a const for `versioninfo`, `pipelines.repositorypropertynames.versioninfo`, i think.

Reply:
PS. VersionInfo is actually a class not a single string. :)